### PR TITLE
Add VilLain

### LINKED ISSUE #171: Add VilLain model
- Paper: https://dl.acm.org/doi/pdf/10.1145/3589334.3645454.
- Code: https://github.com/geon0325/VilLain.

### PR DESCRIPTION
**Pull request by tizianocitro**
### All Submissions

- [x] Have you followed the guidelines in our [Contributing](../../CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### Description

- Added VilLain model.
- Added binary accuracy metric to all examples.
- Added support for maxmin aggreation.
 
### Checklist

- [x] Does your submission pass all tests? (use `make test`)
- [x] Have you written tests to cover all your changes? If not, provide a reason.
- [x] Have you lint your code locally before submission? (use `make lint`)
- [x] Have you type checked your code locally before submission? (use `make typecheck`)
- [x] Have you added an explanation of what your changes are and why you'd like us to include them?

### CODE REVIEW

File: hyperbench/nn/loss.py [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+
+        Returns:
+            Scalar loss value.
+        """
+        # Split logits by label as we need to compare positive scores against negative scores.
+        # Example: logits = [2.0, 1.0, -1.0]
+        #          labels = [1.0, 1.0, 0.0]
+        #          -> positive_logits = [2.0, 1.0]
+        #          -> negative_logits = [-1.0]
+        positive_logits = logits[labels == 1]
```
Comment (ddevin96) [inline hyperbench/nn/loss.py:37]: Check that the comparison between float and integer is always correct

Comment (tizianocitro) [reply hyperbench/nn/loss.py:37]: It works correctly, and also looks like it's the more idiomatic way.

**Review by ddevin96 — APPROVED** [review]

**Review by ddevin96 — APPROVED** [review]
